src/crewai/llm.py

```diff
@@ -54,12 +55,15 @@
 load_dotenv()
 
 
-class FilteredStream:
-    def __init__(self, original_stream):
+class FilteredStream(io.TextIOBase):
+    def __init__(self, original_stream: TextIO):
         self._original_stream = original_stream
         self._lock = threading.Lock()
 
-    def write(self, s) -> int:
+    def write(self, s: str) -> int:
+        if not hasattr(self, "_lock"):
```

Comment:
Nothing to do now, but I don't love these `hasattr` we have in several places. Hope we can refactor these eventually.

Reply:
good call.. just refactored that.

Reply:
Lovely! Thanks!

Once again, nothing to do here, but what do you think about being more descriptive? Results are definitely the same, since the compiler takes care of proper typing.
```
_lock: Optional[threading.Lock] = None # although I still prefer the more modern but still unsupported Lock | None
```